replication/gatys_et_al_2017/main.py

```diff
@@ -0,0 +1,284 @@
+
+        with replicate_figure(args.logger, "3 (d)", args.impl_params):
+            output_luminance = gatys_et_al_2017_nst(
+                content_luminance,
+                style_luminance,
+                impl_params=args.impl_params,
+                quiet=args.quiet,
+                logger=args.logger,
+            )
+            output_luminance = torch.mean(output_luminance, dim=1, keepdim=True)
+            output_chromaticity = resize(
+                content_chromaticity, output_luminance.size()[2:]
+            )
+            output_image_yuv = torch.cat((output_luminance, output_chromaticity), dim=1)
+            output_image = yuv_to_rgb(output_image_yuv)
+
+            output_file = path.join(args.image_results_dir, "fig_3__d.jpg")
+            log_saving_info(args.logger, output_file)
+            write_image(output_image, output_file)
+
+    @abort_if_cuda_memory_exausts
+    def figure_3_e(content_image, style_image, method="cholesky"):
+        style_image = match_channelwise_statistics(style_image, content_image, method)
+
+        with replicate_figure(args.logger, "3 (e)", args.impl_params):
+            output_image = gatys_et_al_2017_nst(
+                content_image,
+                style_image,
+                impl_params=args.impl_params,
+                quiet=args.quiet,
+                logger=args.logger,
+            )
+
+            output_file = path.join(args.image_results_dir, "fig_3__e.jpg")
+            log_saving_info(args.logger, output_file)
+            write_image(output_image, output_file)
+
+    images = gatys_et_al_2017_images()
+    images.download(root=args.image_source_dir)
+    content_image = images["schultenhof"].read(device=args.device)
+    style_image = images["starry_night"].read(device=args.device)
+
+    figure_3_c(content_image, style_image)
+    figure_3_d(content_image, style_image)
+    figure_3_e(content_image, style_image)
+
+
+def parse_input():
+    # TODO: write CLI
+    image_source_dir = None
+    image_guides_dir = None
+    image_results_dir = None
+    device = None
+    impl_params = True
+    quiet = False
+
+    def process_dir(dir):
+        dir = path.abspath(path.expanduser(dir))
+        os.makedirs(dir, exist_ok=True)
+        return dir
+
+    here = path.dirname(__file__)
+
+    if image_source_dir is None:
+        image_source_dir = path.join(here, "images", "source")
+    image_source_dir = process_dir(image_source_dir)
+
+    if image_guides_dir is None:
+        image_guides_dir = path.join(here, "images", "guides")
+    image_guides_dir = process_dir(image_guides_dir)
+
+    if image_results_dir is None:
+        image_results_dir = path.join(here, "images", "results")
+    image_results_dir = process_dir(image_results_dir)
+
+    if device is None:
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+    if isinstance(device, str):
+        device = torch.device(device)
+
```

Comment:
Use `get_device` from `pystiche.misc`

```suggestion
    device = get_device()
```